coralproject/talk
# [CORL-2934]: DSA - Moderating illegal content

## What does this PR do?

These changes add the ability to moderate illegal content. They include the addition of a `Reports` tab to the admin that shows a table of all DSAReports made. When one of these reports is clicked on, a single report page is shown with that report's info, history, and the ability to add notes, download its info, change its status, and make an illegal or not illegal decision.

Still TODO:
- [x] add decision screen for moderating reports
- [x] update copy where needed - (will continue this in future PRs)
- [x] address TODOs in code
- [x] add tests - (will continue this in future PRs)
- [x] add link back to reports table from single report screen
- [x] update CSV file that's downloaded with design input

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

These changes add the ability to query for DSAReports and add to their history (add notes, change status, delete notes, download the report data, make a decision).

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?



## How do I test this PR?

You can ensure that DSA is enabled. Have some DSAReports created. Go to the `Reports` tab in the admin. See the table of DSAReports. See that it can be sorted by newest/oldest. See that you can filter by closed reports. Click into a specific report. See its data. See its history, including when it was submitted. Add a note to its history. Delete a note from its history. Click `Share` button to download a csv file with the DSAReport's data. Change its status. Make a decision on a DSAReport. View its reported comment in the stream/moderation. See the user history drawer open if you click on the reporter username or the commenter's username. 

When you make a decision on a report, if it's a decision that it is illegal, you should see that the comment is rejected.

If you add a note to or download a CSV for a report that is status `Awaiting review` still, you will be prompted if you would like to move the report to `In review` with a modal. If you select `Completed` status for a report, you will be prompted by a modal to make a decision on that report.

To test how this works with GDPR deletion, you can have a user 1 comment, then user 2 report the comment for illegal content, and then user 1 request their account be deleted. Update their scheduled deletion date to something sooner. Run with jobs to have the deletion go through. See that the pending DSA report (must be in `In review` or `Awaiting review` status still) is set to `Void` status, and that you get a note about the reported comment being deleted when viewing the DSA report as well.

## Where any tests migrated to React Testing Library?



## How do we deploy this PR?
